tests/test_reclaim_swap.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+'''
+Tests eos-reclaim-swap.
+'''
+
+import os
+import subprocess
+import tempfile
+
+from .util import (
+    BaseTestCase,
+    system_script
+)
+
+
+EOS_RECLAIM_SWAP = system_script('eos-reclaim-swap')
+DATA_DIR = './data/reclaim-swap'
+
+class TestReclaimSwap(BaseTestCase):
+
+    def get_data(self, basename):
+        '''
+        Return contents of file at DATA_DIR/basename.
+        '''
+        path = os.path.join(DATA_DIR, basename)
+        with open(path, 'rb') as f:
+            data = f.read()
+        return data
+
+
+    def dup_image(self, testname):
+        '''
+        Return a copy of DATA_DIR/testname.img in the current working dir.
+        '''
+        basename = testname + '.img'
+        path = os.path.join(DATA_DIR, basename)
+        with open(path, 'rb') as f:
+            data = f.read()
+        img = open(basename, 'wb')
+        img.write(data)
+        img.flush()
+        return img
+
+
+    def strip_table_id(self, table):
+        '''
+        Return a copy of table without the partition table line.
+        '''
+        lines = table.splitlines(keepends=True)
+        out = []
+        for l in lines:
+            if l.startswith(b'Disk identifier:'):
+                continue
+            out.append(l)
+        return bytes().join(out)
+
+
+    def run_test(self, test):
+        '''
+        Execute test.
+        '''
+        input_table = self.get_data(test + '-input.txt')
+        expected_table = self.get_data(test + '-expected.txt')
+
+        with self.dup_image(test) as img:
+            table = subprocess.run(['fdisk', '-l', img.name],
+                                   stdout=subprocess.PIPE).stdout
+            try:
+                self.assertEqual(table, input_table)
+            except:
+                print('Unexpected input image with table:\n' + table.decode())
+                raise
+
+            subprocess.run([EOS_RECLAIM_SWAP, '-v', '-p', img.name + '3'],
+                           check=True, stderr=subprocess.DEVNULL)
```

Comment:
Why suppress stderr here?

Reply:
I assumed the pyparted import error when running as non-root would impact the test results, but that is not the case. Dropping.